mypy/checkstrformat.py

```diff
@@ -380,8 +380,9 @@ def perform_special_format_checks(self, spec: ConversionSpecifier, call: CallExp
                 if (has_type_component(actual_type, 'builtins.bytes') and
                         not custom_special_method(actual_type, '__str__')):
                     self.msg.fail(
-                        "On Python 3 '{}'.format(b'abc') produces \"b'abc'\", not 'abc'; "
-                        "use '{!r}'.format(b'abc') if this is desired behavior",
+                        'On Python 3 formatting string with "{}" and bytes argument '
```

Comment:
The new wording is harder to understand. I'd suggest something closer to the original wording but using an f-string, like this:
```suggestion
                        "On Python 3 f'{b"abc"}' produces \"b'abc'\", not 'abc'; "
```
However, it looks like the same code path is used for f-strings and `.format()`. Is it easy to use separate messages for each of these?

Reply:
> Is it easy to use separate messages for each of these?

Currently `mypy` uses the same exact `ast` to represent both `.format` and `f` strings. I can add new metadata to `CallExpr` (which is used for representation). But, I've decided that simply changing an error message is the way to go.

Maybe you can come up with better wording?

Reply:
Then I suggest we just always use the f-string version of the message if we're type checking Python 3, since I would assume that's the more likely case. Mypy doesn't support 3.5 any more. If this message is triggered when we're checking Python 2 code (can it?) we should still mention `.format()`.

Reply:
> Then I suggest we just always use the f-string version of the message if we're type checking Python 3, since I would assume that's the more likely case.

I cannot agree, for example in my projects `f` string is not allowed by our styleguide: https://wemake-python-stylegui.de/en/latest/pages/usage/violations/consistency.html#wemake_python_styleguide.violations.consistency.FormattedStringViolation So, I know that a lot of people still use `.format()` 🙂 

But, we can use the common part like `{}` and `{!r}`. Maybe something like

```
On Python 3 '{b"abc"}' produces \"b'abc'\", not 'abc'; use "{b"abc"!r}" if this is desired behavior
```